### ISSUE TITLE
why do the loss equal nan when i don't use clipnorm ?

### ISSUE THREAD

**Issue by ghost** [fchollet/keras#5127]
i know that clipnorm fix this issue and i know that clipnorm clip the big number of the gradients but i want to know why the nan is produced?
why do i see loss=nan when i don't use clipping of the gradient?

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by ncullen93** [comment]
the gradient gets really really large, which changes the weight values a LOT... and this makes the loss NaN because it's usually too large of a number for python to represent.

**Comment by Bassel1** [comment]
but what is the maximum number python can handle
i googled it and found 
`
import sys;  
print(sys.maxsize);  
9223372036854775807`

i mean, may you tell me an example of that big number in loss that make it show NaN?
